agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -758,21 +756,27 @@ func MetadataFromContainer(dockerContainer *docker.Container) DockerContainerMet
 			return DockerContainerMetadata{Error: apierrors.NamedError(err)}
 		}
 	}
+
+	createdTime, _ := time.Parse(time.RFC3339Nano, dockerContainer.Created)
+	startedTime, _ := time.Parse(time.RFC3339Nano, dockerContainer.State.StartedAt)
+	finishedTime, _ := time.Parse(time.RFC3339Nano, dockerContainer.State.FinishedAt)
+	volumeMap := mountPoints(dockerContainer.Mounts)
+
 	metadata := DockerContainerMetadata{
 		DockerID:     dockerContainer.ID,
 		PortBindings: bindings,
-		Volumes:      dockerContainer.Volumes,
-		CreatedAt:    dockerContainer.Created,
-		StartedAt:    dockerContainer.State.StartedAt,
-		FinishedAt:   dockerContainer.State.FinishedAt,
+		Volumes:      volumeMap,
+		CreatedAt:    createdTime,
+		StartedAt:    startedTime,
+		FinishedAt:   finishedTime,
 	}
 	if dockerContainer.Config != nil {
 		metadata.Labels = dockerContainer.Config.Labels
 	}
 
 	metadata = getMetadataVolumes(metadata, dockerContainer)
```

Comment:
Since you are setting metadata's volumes field using `mountPoints` method here https://github.com/aws/amazon-ecs-agent/pull/1530/files#diff-d50c02a5fb063b2c12c140e966c2b710R763, this can be removed(and `getMetadataVolumes` too)